tools/testing/target_determination/determinator.py

```diff
@@ -0,0 +1,58 @@
+from typing import List
+
+from tools.stats.upload_stats_lib import emit_metric
+
+from tools.testing.target_determination.heuristics import (
+    HEURISTICS,
+    TestPrioritizations,
+)
+
+
+def _print_tests(label: str, tests: List[str]) -> None:
+    if not tests:
+        return
+
+    print(f"{label} tests:")
+    for test in tests:
+        if test in tests:
+            print(f"  {test}")
+
+
+def get_test_prioritizations(tests: List[str]) -> TestPrioritizations:
+    rankings = TestPrioritizations()
+    rankings.unranked_relevance = tests
+
+    for heuristic in HEURISTICS:
+        new_rankings = heuristic.get_test_priorities(tests)
+        rankings.integrate_priorities(new_rankings)
+
+        num_tests_found = len(new_rankings.highly_relevant) + len(
+            new_rankings.probably_relevant
+        )
+        print(
+            f"Heuristic {heuristic} identified {num_tests_found} tests \
+              to prioritize ({(num_tests_found / len(tests)):.2%}%)"
+        )
+
+        if num_tests_found:
+            _print_tests("Highly relevant", new_rankings.highly_relevant)
+            _print_tests("Probably relevant", new_rankings.probably_relevant)
+
+    num_tests_analyzed = (
+        len(rankings.highly_relevant)
+        + len(rankings.probably_relevant)
+        + len(rankings.unranked_relevance)
+    )
+    assert num_tests_analyzed != len(
```

Comment:
Could you briefly explain what is the goal of this assert?  I would naively expect `num_tests_analyzed` to be equal to the length of all input tests, but the assert says otherwise.

Reply:
You're right. This was a buggy assert and I just realized I'd accidentally put the fix for this [in a PR](https://github.com/pytorch/pytorch/pull/107071/files#diff-167301b680ef3cd2de2c553db17e6a6c9e46936d3beb1e689149691cbca3b93dL46) higher up in the stack